src/main/java/com/google/sps/servlets/UserServlet.java

```diff
@@ -90,30 +109,45 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
     response.setContentType("application/json; charset=UTF-8");
     response.setCharacterEncoding("UTF-8");
 
-    String json = convertToJson(user);
-    response.getWriter().println(json);
+    response.getWriter().println(user.toJSON());
   }
 
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
-      UserService userService = UserServiceFactory.getUserService();
-      String userId = userService.getCurrentUser().getUserId();
-      String userNickname = request.getParameter(UserInfo.NICKNAME);
+    GoogleIdToken idToken = verifyId(request.getParameter(ID_TOKEN_PARAM));
 
```

Comment:
nit: Improve readability by keeping the error control flow logic similar between `doGet`, `doPost`, and `doPut`. In this case, you'd return a 400 if (idToken == null) here, like you do in lines 88-92 and 143-147 of this file.